src/controllers/PartialSubmissionController.php

```diff
@@ -166,29 +174,37 @@ protected function savePartialFile($formData, array $filter, EditableFormField\E
             $partialFileSubmission = PartialFileFieldSubmission::create($partialData);
             $partialFileSubmission->write();
         }
-        // Don't overwrite existing uploads
-        if (!$partialFileSubmission->UploadedFileID && is_array($formData['Value'])) {
-            $file = $this->uploadFile($formData, $editableField);
+
+        if (is_array($formData['Value'])) {
+            $file = $this->uploadFile($formData, $editableField, $partialFileSubmission);
             $partialFileSubmission->UploadedFileID = $file->ID ?? 0;
+            $partialFileSubmission->write();
         }
-        $partialFileSubmission->write();
     }
 
     /**
      * @param array $formData
      * @param EditableFormField\EditableFileField $field
+     * @param PartialFileFieldSubmission $partialFileSubmission
      * @return bool|File
-     * @throws \Exception
+     * @throws Exception
      */
-    protected function uploadFile($formData, $field)
+    protected function uploadFile($formData, $field, $partialFileSubmission)
     {
         if (!empty($formData['Value']['name'])) {
             $foldername = $field->getFormField()->getFolderName();
 
-            // create the file from post data
+            if (!$partialFileSubmission->UploadedFileID) {
+                $file = File::create([
```

Comment:
Partial Files are being created as public files which is a security risk.
Normal userform submissions create a draft file

Reply:
This is actually determined by each individual EditableFileField, where you can set the file to upload to a secure folder. Not an issue.